code/game/machinery/computer/buildandrepair.dm

```diff
@@ -385,8 +385,7 @@
 							return
 						playsound(get_turf(src), 'sound/items/Welder.ogg', 50, 1)
 						user.visible_message("[user] welds the frame back into metal.", "You weld the frame back into metal.", "You hear welding.")
-						var/obj/item/stack/sheet/metal/M = getFromPool(/obj/item/stack/sheet/metal, src.loc)
-						M.amount = 5
+						new /obj/item/stack/sheet/metal/(src.loc, 5)
```

Comment:
You could locate to see if there already was a stack and manually add to the amount I guess.